snap/snapcraft.yaml

```diff
@@ -58,6 +58,22 @@ parts:
       mkdir -p $SNAPCRAFT_PART_INSTALL/opt/powershell
       tar zxf powershell.tar.gz -C $SNAPCRAFT_PART_INSTALL/opt/powershell
       chmod +x $SNAPCRAFT_PART_INSTALL/opt/powershell/pwsh
+      echo "building third party notices file - $thirdPartNoticeFile"
+      thirdPartNoticeFile="$SNAPCRAFT_PART_INSTALL/thirdPartyNotices.txt"
+      echo 'THIRD-PARTY SOFTWARE NOTICES AND INFORMATION' > $thirdPartNoticeFile
```

Comment:
A heredoc might suit this situation well:
```sh
cat > $thirdPartyNoticeFile <<EOF
THIRD-PARTY SOFTWARE NOTICES AND INFORMATION
This software is based on or incorporates material from the projects listed below (collectively, "Third Party Code").
Microsoft is not the original author of the Third Party Code.
The original copyright notice and license, under which Microsoft received such Third Party Code, are set forth below.
Microsoft reserves all rights not expressly granted herein, whether by implication, estoppel or otherwise.
EOF
```